### PR TITLE
chore(avaxc) : wrw support for evm ccr 

### PR DESCRIPTION
**Pull request by nrjsuthar**
chore(avaxc) : wrw support for evm ccr 

TICKET: COIN-1708

### CODE REVIEW

**Comment by kamleshmugdiya** [comment]
tests are failing

File: modules/sdk-coin-avaxc/src/EthLikeAvaxc.ts [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+  async getGasPriceFromExternalAPI(): Promise<BN> {
+    try {
+      // COIN -1708 : hardcoded for half signing
+      const gasPrice = new BN(250000);
+      console.log(` Got hardcoded gas price: ${gasPrice}`);
+      return gasPrice;
+    } catch (e) {
+      throw new Error('Failed to get gas price');
+    }
+  }
```
Comment (kamleshmugdiya) [inline modules/sdk-coin-avaxc/src/EthLikeAvaxc.ts:338]: same

**Review by kamleshmugdiya — CHANGES_REQUESTED** [review]
this defeats the purpose of inheritance and code reusability.

**Comment by nrjsuthar** [comment]
@kamleshmugdiya 
This PR https://github.com/BitGo/BitGoJS/pull/4925 had the change you mentioned, where the relationship exists. The changes in the current PR were because of the Gas and Limit, wherein all the parent methods are copied. The ideal change would be to extend avaxc class with abstractEthLikeNewCoin (mentioned PR), but removed as no one was sure of the change.

**Comment by nrjsuthar** [comment]
Also, the gas price and limit from the explorer were not required here for the half sign.

File: modules/sdk-coin-avaxc/src/EthLikeAvaxc.ts [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+
+    const gasPriceMax = ethGasConfigs.maximumGasPrice;
+    const gasPriceMin = ethGasConfigs.minimumGasPrice;
+    if (userGasPrice < gasPriceMin || userGasPrice > gasPriceMax) {
+      throw new Error(`Gas price must be between ${gasPriceMin} and ${gasPriceMax}`);
+    }
+    return userGasPrice;
+  }
+
+  async recoveryBlockchainExplorerQuery(query: Record<string, any>): Promise<any> {
```
[the remark this answers is not in the document]
Comment (nrjsuthar) [reply modules/sdk-coin-avaxc/src/EthLikeAvaxc.ts:62]: ok

**Comment by kamleshmugdiya** [comment]
> @kamleshmugdiya This PR #4925 had the change you mentioned, where the relationship exists. The changes in the current PR were because of the Gas and Limit, wherein all the parent methods are copied. The ideal change would be to extend avaxc class with abstractEthLikeNewCoin (mentioned PR), but removed as no one was sure of the change.

I don't understand this, if the function has the same impl in AbstractEthLikeNewCoins, what is the need to override with the same code in the new class?

**Comment by nrjsuthar** [comment]
> > @kamleshmugdiya This PR #4925 had the change you mentioned, where the relationship exists. The changes in the current PR were because of the Gas and Limit, wherein all the parent methods are copied. The ideal change would be to extend avaxc class with abstractEthLikeNewCoin (mentioned PR), but removed as no one was sure of the change.
> 
> I don't understand this, if the function has the same impl in AbstractEthLikeNewCoins, what is the need to override with the same code in the new class?

This was based on testing, for each of the issues, the method has been copied. 

Example:

A -> B -> C is the method flow and A has to be changed, now i we need to change the method A over here, we will need to override all its parents, the sub method will not be able to be called.

**Comment by kamleshmugdiya** [comment]
> > > @kamleshmugdiya This PR #4925 had the change you mentioned, where the relationship exists. The changes in the current PR were because of the Gas and Limit, wherein all the parent methods are copied. The ideal change would be to extend avaxc class with abstractEthLikeNewCoin (mentioned PR), but removed as no one was sure of the change.
> > 
> > 
> > I don't understand this, if the function has the same impl in AbstractEthLikeNewCoins, what is the need to override with the same code in the new class?
> 
> This was based on testing, for each of the issues, the method has been copied.
> 
> Example:
> 
> A -> B -> C is the method flow and A has to be changed, now i we need to change the method A over here, we will need to override all its parents, the sub method will not be able to be called.

what testing?
and the methods that I pointed out, do they have any difference in implementation vs the AbstractEthLikeNewCoins?
they all looked the same to me.

File: modules/sdk-coin-avaxc/src/avaxc.ts [hunk 3]
```diff
@@ -62,20 +62,21 @@ import {
   VerifyAvaxcTransactionOptions,
 } from './iface';
 import { AvaxpLib } from '@bitgo/sdk-coin-avaxp';
+import { EthLikeAvaxc } from './EthLikeAvaxc';
 
 export class AvaxC extends BaseCoin {
   static hopTransactionSalt = 'bitgoHopAddressRequestSalt';
-
   protected readonly _staticsCoin: Readonly<StaticsBaseCoin>;
+  private ethLikeAvaxc: any;
```
Comment (kamleshmugdiya) [inline modules/sdk-coin-avaxc/src/avaxc.ts:70]: 1. Why is composition relationship required?
2. can you name this better EthLikeAvaxc?
just by name, I am not able to understand what is the diff between EthLikeAvaxc and AvaxC.

**Comment by kamleshmugdiya** [comment]
pipeline failing again

**Pull request closed without merging**